components/storage/blockdevice/COMPONENT_FLASHIAP/FlashIAPBlockDevice.cpp

```diff
@@ -256,4 +256,13 @@ const char *FlashIAPBlockDevice::get_type() const
     return "FLASHIAP";
 }
 
+bool FlashIAPBlockDevice::is_valid_erase(bd_addr_t addr, bd_size_t size) const
+{
+    bd_addr_t base_addr = addr + (_base - _flash.get_flash_start());
```

Comment:
Please add a comment saying what is base_addr and that it is related to the full flash, not only the addresses inside the blockdevice